src/serial.rs

```diff
@@ -1442,6 +1453,27 @@ macro_rules! halUsartImpl {
                             })
                     });
 
+                    match config.dma {
+                        DmaConfig::Tx => {
+                            usart.cr3.write(|w| {
+                                w.dmat().enabled()
+                            })
+                        }
+                        DmaConfig::Rx => {
+                            usart.cr3.write(|w| {
+                                w.dmar().enabled()
+                            })
+                        }
+                        DmaConfig::TxRx => {
+                            usart.cr3.write(|w| {
+                                w
+                                    .dmar().enabled()
+                                    .dmat().enabled()
+                            })
+                        }
+                        _ => {}
```

Comment:
Maybe this should be `DmaConfig::None` instead ?

Reply:
Ah, good point. I'll modify.

wrt hardware, yes, on an stm32f401re against my USART6.